contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -34,6 +34,11 @@ struct trilogy_ctx {
     VALUE encoding;
 };
 
+struct trilogy_result_columns_ctx {
+    struct column_info *column_info;
+    uint64_t column_count;
+};
```

Comment:
```suggestion
struct trilogy_result_columns_ctx {
    uint64_t column_count;
    struct column_info column_info[];
}; 
```

You could use a flexible array member to allocate the whole list at once and avoid pointer chasing and malloc churn.